### ISSUE TITLE
1.8.9 Support

### ISSUE THREAD

**Issue by Dreestar** [LeviDevs/LunarClientSpoofer#5]
Hello.
The mod is really cool. Thanks for the work.
I think a lot of people would be happy if it will work for version 1.8.9.
Are you planning to add it or not ?

Have a nice day / night!

**Comment by leafypout** [comment]
Hi, a lot of people have been requesting 1.8 support for at least the path month or so. I am finally working on it, there is a new branch for it as it's not ready yet which can be found below. (It currently doesn't work on 1.8 [it just crashes] but when it does it'll be on the master branch with a new release.)
https://github.com/LeviDevs/LunarClientSpoofer/tree/mc-1.8.9

**Comment by glamingles** [comment]
Glad to hear :D. Just wondering, if you could make a lunar client spoof that spoofs the anti cheat, would you? I know a lot of people would be interested in it because of the major fps drop some people get with lunar. A lot of us want to play on the network (ranked) but it requires a client. I understand that you're very busy, but do you know anyone that could make this, comission wise? Sorry for bothering you again, thank you.

**Comment by leafypout** [comment]
They removed the anti-cheat functionality like 2-3 days ago, the anti-cheat is no longer provided in Lunar Client.

That being said, once this mod is "done" with the logo spoofing, it will basically be Lunar Client. If they add the anti-cheat back, it will be a different story.

**Comment by glamingles** [comment]
That is true, but to play ranked on Lunar you still need to be on Lunar client for some reason. I can't play it on BLC or Forge.

**Comment by glamingles** [comment]
Oh, I was wrong, you can play on forge and stuff. Thanks for the information.

**Comment by Dreestar** [comment]
> 
> 
> They removed the anti-cheat functionality like 2-3 days ago, the anti-cheat is no longer provided in Lunar Client.
> 
> That being said, once this mod is "done" with the logo spoofing, it will basically be Lunar Client. If they add the anti-cheat back, it will be a different story.

Thank you so much for your effort!

How much longer do you need until the release?

Have a good day / night :)

**Comment by matthieu-ctrl** [comment]
plsplspls do it

**Comment by glamingles** [comment]
Any updates? I tried porting the mod myself and it did not go well haha

**Comment by leafypout** [comment]
I have basically got a working version for 1.8.9 but there are several issues with having it work for both 1.7.10 and 1.8.9, as well as that Forge seems to have broken 1.7.10 a few weeks ago (could just be me, not sure.)

I won't be working on this any time soon and I cannot guarantee I will make a version for 1.8.9. If I do, they will most likely be different versions (ie multiple versions of the mod for each MC version.) Making something like this cross-compatible is a real pain.

Regarding the logo:
Lunar Client sends your Minecraft account's auth token to their servers to "verify" you're on Lunar Client. If I ever add support for logos it will be disabled by default to protect user's privacy (Lunar is already breaching user's privacy by doing this, and who knows what they're doing with your auth tokens.)


Now that Lunar has opened up to any client since their anti-cheat no longer exists, I don't see much of a point in this mod. Here are my opinions:
- If you want to cheat on Lunar and have the Lunar Client logo just use a client that injects into Lunar. (I believe Vape and several other clients do this just fine now)
- If you want to cheat but don't care about the logo, do whatever you want.
- Otherwise, just play on Lunar Client. Unfortunately this is kind of an ultimatum here; if you don't send your auth token to their servers (ie by blocking the URL it sends it to in Windows Firewall or similar) then the logo won't be shown above your head.

Regarding auth tokens: I believe they send it encrypted over TLS so you should be fine, however what they do with it I have no idea. I personally don't trust Lunar with my information but everyone can form their own opinions. As far as I know they just use it so you can't spoof the logo state for other players but since it's being sent to their servers, they could be doing literally anything with them.

**Comment by glamingles** [comment]
Thanks for this response. I never knew about how the logos works and this is some insightful information. 

Someone did make a 1.8 version (although it seems to be broken), not sure if its because of the same issues you have run into.
https://hypixel.net/threads/fixed-forge-1-8-9-lunarspoof.3484115/.

 I honestly don't think you should care about making cross-compatible, since people that want it on 1.8 most likely want it for hypixel which disallows 1.7. 

Mainly for me Lunar Client doesn't work on my PC, no idea why and have done vigorous research onto how to fix this, admin didn't give a good response. The spoofer allows me to play on the lunar server (ranked (you can't go on forge or badlion client to play ranked), but they recently switched to 1.8 only.

Honestly, i'm not going to try to play their server anymore - just a bunch a bs consistently with lag and stuff like this. 

Thanks for your detailed responses and making this mod in the first place.

**Comment by leafypout** [comment]
Honestly the mod in the Hypixel link looks pretty shit and I doubt it runs decently. It looks like he's doing the same thing I'm doing but using some mixins BS because he doesn't know how ASM works.

I wasn't aware of Lunar being 1.8 only; is this true? If so there is no need for 1.7 support. Let me know.

**Comment by glamingles** [comment]
Lunar recently disabled 1.7. You can’t join their server with 1.7 for some
reason. It’s one of the main reasons mmc is gaining popularity.

So yeah, so people that have trouble launching Lunar Client and want to
play ranked or what not are turning towards mods like yours.

**Comment by glamingles** [comment]
I did some more researching and found that if you use the current mod on 1.7, 1.8 users will see the Lunar Client logo next to your name.

With this in mind, would it be possible to make an 1.8, exclusive, non cross-version of the mod? 

Sorry again to bother you

**Comment by unoriginals** [comment]
Can you release the working 1.8.9 version you have?

**Comment by leafypout** [comment]
> I did some more researching and found that if you use the current mod on 1.7, 1.8 users will see the Lunar Client logo next to your name.
> 
> With this in mind, would it be possible to make an 1.8, exclusive, non cross-version of the mod?
> 
> Sorry again to bother you

This is quite interesting. I imagine it only works this way on Lunar and (possibly) servers using their official API (partnered servers.) Normally, you have to send a HTTP request to some endpoint of theirs and then it will work on all servers like Hypixel etc.



> Can you release the working 1.8.9 version you have?

Will some time within the next few days

**Comment by unoriginals** [comment]
> > I did some more researching and found that if you use the current mod on 1.7, 1.8 users will see the Lunar Client logo next to your name.
> > With this in mind, would it be possible to make an 1.8, exclusive, non cross-version of the mod?
> > Sorry again to bother you
> 
> This is quite interesting. I imagine it only works this way on Lunar and (possibly) servers using their official API (partnered servers.) Normally, you have to send a HTTP request to some endpoint of theirs and then it will work on all servers like Hypixel etc.
> 
> > Can you release the working 1.8.9 version you have?
> 
> Will some time within the next few days

https://github.com/LeviDevs/LunarClientSpoofer/tree/mc-1.8.9 is a dead link for me

**Comment by ghost** [comment]
> > I did some more researching and found that if you use the current mod on 1.7, 1.8 users will see the Lunar Client logo next to your name.
> > With this in mind, would it be possible to make an 1.8, exclusive, non cross-version of the mod?
> > Sorry again to bother you
> 
> This is quite interesting. I imagine it only works this way on Lunar and (possibly) servers using their official API (partnered servers.) Normally, you have to send a HTTP request to some endpoint of theirs and then it will work on all servers like Hypixel etc.
> 
> > Can you release the working 1.8.9 version you have?
> 
> Will some time within the next few days

Could you please upload this

**Comment by leafypout** [comment]
I have a working 1.8 equivalent however I have not yet tested it. I'm not currently working on anything related to Minecraft, I'll get around to it when I'm interested again. Pull requests are welcome if someone wants to make the necessary changes required for 1.8.9.

**Comment by Stimpy32** [comment]
Can You update it to 1.8.9 it will be really helpful to a lot of people

**Comment by kowexe** [comment]
> I have a working 1.8 equivalent however I have not yet tested it. I'm not currently working on anything related to Minecraft, I'll get around to it when I'm interested again. Pull requests are welcome if someone wants to make the necessary changes required for 1.8.9.

Can you send it now?

**Comment by rafalohaki** [comment]
bump
